Framework/Kernel/src/NexusHDF5Descriptor.cpp

```diff
@@ -219,4 +228,31 @@ NexusHDF5Descriptor::initAllEntries() {
   return allEntries;
 }
 
+bool NexusHDF5Descriptor::isEntry(const std::string &entryName,
+                                  const std::string &groupClass) const
+    noexcept {
+
+  auto itClass = m_allEntries.find(groupClass);
+  if (itClass == m_allEntries.end()) {
+    return false;
+  }
+
+  if (itClass->second.count(entryName) == 1) {
+    return true;
+  }
+
+  return false;
+}
+
+bool NexusHDF5Descriptor::isEntry(const std::string &entryName) const noexcept {
+
+  for (auto itClass = m_allEntries.cbegin(); itClass != m_allEntries.cend();
+       ++itClass) {
+    if (itClass->second.count(entryName) == 1) {
+      return true;
+    }
+  }
```

Comment:
range-based for loop?

```suggestion
  for (const auto &class: m_allEntries) {
    if (class.second.count(entryName) == 1) {
      return true;
    }
  }
```

Reply:
The loop is doing a reverse search (`cbegin`, `cend`). The majority of variables live in the lower levels of the `std::map` so it's more likely to start searching at the end.

Reply:
Would a reverse search be `rbegin` and `rend`?

Reply:
You're correct. I totally mixed `c` with `r`, my bad. Thanks!

Reply:
Now I wonder if I screwed in other parts of the PR, will look. Thanks!